openapi_python_client/parser/openapi.py

```diff
@@ -245,6 +245,8 @@ def _add_parameters(
             if param.param_in == oai.ParameterLocation.QUERY:
                 endpoint.query_parameters.append(prop)
             elif param.param_in == oai.ParameterLocation.PATH:
+                if not param.required:
+                    return ParseError(data=param, detail="Path parameter must be required"), schemas
```

Comment:
Too bad Pydantic doesn't support tagged unions or we could just stick this in the parsing schemas.